statistics/statistics_test.go

```diff
@@ -105,25 +104,19 @@ func TestStatisticsProcessorMetrics(t *testing.T) {
 
 			movingAverageObj := New()
 
-			_, receivedData, _ := movingAverageObj.Process("snap.gob", buf.Bytes(), nil)
-
-			var metricsNew []plugin.PluginMetricType
+			movingAverageObj.Process("snap.gob", buf.Bytes(), nil)
 
-			//Decodes the content into MetricType
-			dec := gob.NewDecoder(bytes.NewBuffer(receivedData))
-			dec.Decode(&metricsNew)
+			var metricsNew []plugin.MetricType
 			So(metrics, ShouldNotResemble, metricsNew)
```

Comment:
if we can have mock tests to verify the actual vs expected, that would be awesome.